File: voghion_flutter_module/packages/vg_native_mix_flutter/ios/Classes/VgFlutterNavigator.swift
//
//  HqFlutterNavigator.swift
//  hq_navite_mix_flutter
//
//  Created by edy on 2025/7/7.
//

import UIKit

 @objc public enum VgPageType:Int {
    case navite = 1
    case flutter = 2
}


@objc public class VgFlutterNavigator: NSObject {
    @objc public static let shared = VgFlutterNavigator()
    // 私有构造函数防止外部初始化
    private override init() {}
    
    @objc public weak var delegate:VgNavigatorProtocol?
    
    @objc public func openPage(routeName:String,routeArgs: Any? = nil,pageType:VgPageType) {
        
        if pageType == .navite {
            self.delegate?.openNativePage(routeName: routeName, args: routeArgs)
            return
        }
        if pageType == .flutter {
            self.delegate?.openFutterPage(routeName: routeName, args: routeArgs)
            return
        }
        
    }
    
    @objc public func refreshFlutterPage(routeName:String,routeArgs: Dictionary<String, Any>? = nil) {
        let channel = VgNativeMixFlutterPlugin.shared?.methodChannel
        channel?.invokeMethod("refreshPage",arguments: routeArgs)
        
    }
}

File: voghion_flutter_module/packages/vg_native_mix_flutter/android/src/main/kotlin/com/voghion/vg_native_mix_flutter/VgFlutterNavigator.kt
package com.voghion.vg_native_mix_flutter

class VgFlutterNavigator {

    companion object {
        val  shared = VgFlutterNavigator()
    }
    var delegate: VgNavigatorInterface? = null

    fun openPage(routeName:String, routeArgs: Map<String, Any?>?, pageType:VgPageType) {
        if (pageType == VgPageType.native) {
            this.delegate?.openNativePage(routeName, args = routeArgs)
            return
        }
        if (pageType == VgPageType.flutter) {
            this.delegate?.openFlutterPage(routeName, args = routeArgs)
            return
        }
    }
     fun refreshFlutterPage(routeName:String,routeArgs: Any? = null) {
         val channel = VgNativeMixFlutterPlugin.shared.methodChannel
         channel.invokeMethod("refreshPage",routeArgs)

    }
}